src/base/bittorrent/sessionimpl.cpp

```diff
@@ -1253,12 +1255,12 @@ int SessionImpl::globalMaxSeedingMinutes() const
 
 void SessionImpl::setGlobalMaxSeedingMinutes(int minutes)
 {
-    if (minutes < 0)
-        minutes = -1;
+    minutes = std::max(minutes, BitTorrent::Torrent::NO_SEEDING_TIME_LIMIT);
```

Comment:
All the clamping should be done here (before comparing with current value):
```suggestion
    minutes = std::clamp(minutes, Torrent::NO_SEEDING_TIME_LIMIT, Torrent::MAX_SEEDING_TIME);
```

Reply:
Done